### DISCUSSION TITLE
Filtering results of diagnostic picker?

### DISCUSSION THREAD

**Discussion by SpaceMacGyver** [helix-editor/helix#11994] [Q&A]
While less than ideal, my code (Rust) has several warnings identified by the diagnostic picker.  Generally, though, if I'm opening the picker, it's because I'm interested in errors, which are often difficult to track down in the midst of several warnings.

The picker seems to understand the distinction between info/warn/error (at least, it presents the white/yellow/red color representations), but I can't figure out how to filter by these different levels.  Does such a mechanism exist?

**Comment by the-mikedavis** [comment]
After the last release there was a major refactor for pickers like the diagnostics picker so that they are laid out like a table and there's a new syntax for querying the columns: https://docs.helix-editor.com/master/pickers.html#filtering-picker-results (also see #9647). So if you build from master you can use a query `%severity error` (or `%s e` for short) to filter the "severity" column for errors.
